packages/message-manager/src/utils.ts

```diff
@@ -112,6 +114,30 @@ export function validateTypedSignMessageDataV3(
       'Data must conform to EIP-712 schema. See https://git.io/fNtcx.',
     );
   }
+
+  if (!currentChainId) {
+    throw new Error('Current chainId cannot be null or undefined.');
+  }
+
+  let { chainId } = data.domain;
+  if (chainId) {
+    if (typeof chainId === 'string') {
+      chainId = parseInt(chainId, chainId.startsWith('0x') ? 16 : 10);
+    }
+
+    const activeChainId = parseInt(currentChainId, 16);
+    if (Number.isNaN(activeChainId)) {
+      throw new Error(
+        `Cannot sign messages for chainId "${chainId}", because MetaMask is switching networks.`,
```

Comment:
Interesting; so this condition won't actually get triggered when the user is switching networks. This didn't work properly for the old code either. Likely this condition has been unreachable for years.

We can check if the network is loading by referencing the `networkStatus` property of the network controller state.

Since this is an older pre-existing issue, maybe we can solve it in a later PR.

Reply:
Thanks for this comment @Gudahtt 👍 
To be honest, it was also unclear for me but I didn't want to act on it. I've created a [backlog item](https://github.com/orgs/MetaMask/projects/47/views/1?pane=issue&itemId=27443634), and will fill it out once this issue is fixed.